Lib/venv/__init__.py

```diff
@@ -303,8 +303,11 @@ def setup_python(self, context):
             copier(context.executable, path)
             if not os.path.islink(path):
                 os.chmod(path, 0o755)
-            for suffix in ('python', 'python3',
-                           f'python3.{sys.version_info[1]}'):
+
+            suffixes = ['python', 'python3', f'python3.{sys.version_info[1]}']
+            if sys.version_info[:2] == (3, 14):
+                suffixes.append('pythonπ')
```

Comment:
I think using `python-pi` or `python_pi` would be better. I think using special characters may cause garbled characters on some platform.

Reply:
Hmm, what platform / filesystem are you concerned about, and what would the impact to the user on that platform be? (note in this branch we know `os.name == "posix"`)

A Greek letter isn't really that special. It's fully normalised Unicode (non-normalised Unicode can still be a little surprising). Looks like even FAT has supported this since around the time I was born via LFN.

This is decidedly one of the advantages of being alive in 2024 :-)

Reply:
Linux console with no framebuffer?

Console fonts without this glyph?  I sometimes do math with the Diofant in the framebuffer console (yes, it has unicode support).  I works well with greek letters, do some pretty-printing to display formulas.  But I doubt it's able to display e.g. chinese texts an non-garbadge.  (Perhaps, with other font, which I don't have...)

Reply:
https://wiki.archlinux.org/title/Linux_console says that the Linux console uses UTF-8, but we may indeed have to worry about whether the console font is missing π. The same page says the kernel built-in font uses the CP437 character set, which does have π (and dates back to the original IBM PC).

It's possible I misunderstood something, but even the case where the font is missing the glyph, is the worst thing that can happen that a user will see an unrendered glyph if ls-ing? That doesn't seem that bad, and sounds in line with what PEP 11 describes as "won't fix".

Of course, if we see any problems during prerelease we should revert.

Reply:
Terminals cope with missing glyphs already (usually by rendering an ASCII `?`, or a dedicated replacement character).

If the active encoding is sufficiently wrong, they don't even see the raw unicode:

```
$ ls -d _[bd]*
_build🐸  _deploy🐸
$ LANG=C ls -d _[bd]*
'_build'$'\360\237\220\270'  '_deploy'$'\360\237\220\270'
```

So yeah, as far as easter eggs go, this one is as close to genuinely harmless as they get.